core/registry/index.go

```diff
@@ -137,3 +137,18 @@ func (m *MicroServiceInstance) WithAppID(v string) *MicroServiceInstance {
 	m.Metadata[common.BuildinTagApp] = v
 	return m
 }
+
+// GetEndPointByProto get endpoint by protocol
+func (m *MicroServiceInstance) GetEndPointByProto(proto string) (string, bool) {
```

Comment:
limit user k8s service port name naming convention, not use code to handle complicated scenario